### PR TITLE
Add mode 'head' and 'libtpu-only' mode for installation script

### PR DESCRIPTION
**Pull request by tonyjohnchen**
'head' mode will install jax, jaxlib, and custom libtpu from head.
'libtpu-only' mode will only install custom libtpu without re-install jax,jaxlib from head. (Noted: build jaxlib from head will take about 10 mins)

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -17,32 +17,72 @@
 # Enable "exit immediately if any command fails" option
```
Comment (rwitten) [inline setup.sh:17]: Can you put a comment at the top describing how to use the module? Looks great otherwise!

**Review by rwitten — CHANGES_REQUESTED** [review]

File: setup.sh (hunk 1 above)
Comment (tonyjohnchen) [reply setup.sh:38]: Done.

I will add a README with more details for internal developer next.

**Review by rwitten — CHANGES_REQUESTED** [review]
Just want to give clearer feedback for how to write the documenation.

File: setup.sh [hunk 2]
```diff
@@ -14,35 +14,96 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# How to use:
```
[the remark this answers is not in the document]
Comment (tonyjohnchen) [reply setup.sh]: Makes sense!
Done.

**Review by rwitten — APPROVED** [review]

**Pull request merged**